File: python-version/family/son.py
"""
Son class - represents a son in the family
"""

from .child import Child
from .person import Gender
from .father import Father
from .mother import Mother


class Son(Child):
    """Represents a son in the family"""
    
    def __init__(self, name: str, father: Father, mother: Mother):
        """Initialize a son"""
        super().__init__(name, father, mother, Gender.MALE)
    
    def describe_siblings(self):
        """Describe the son's siblings"""
        if not self.siblings:
            print(f"{self.name} has no siblings.")
        else:
            for sibling in self.siblings:
                if sibling.gender == Gender.FEMALE:
                    print(f"{self.name} is a {self.get_age()}-year old brother")
                else:
                    print(f"{self.name} is a {self.get_age()}-year old brother")

File: typescript-version/src/family/son.ts
import { Child } from './child';
import { Gender } from './person';
import { Father } from './father';
import { Mother } from './mother';

/**
 * Represents a son in the family
 */
export class Son extends Child {
  constructor(name: string, father: Father, mother: Mother) {
    super(name, father, mother, Gender.MALE);
  }

  /**
   * Describe the son's siblings
   */
  describeSiblings(): void {
    if (this.siblings.length === 0) {
      console.log(`${this.name} has no siblings.`);
    } else {
      for (const sibling of this.siblings) {
        if (sibling.gender === Gender.FEMALE) {
          console.log(`${this.name} is a ${this.getAge()}-year old brother`);
        } else {
          console.log(`${this.name} is a ${this.getAge()}-year old brother`);
        }
      }
    }
  }
}
